hummingbot/connector/exchange/beaxy/beaxy_api_order_book_data_source.py

```diff
@@ -42,25 +41,62 @@ def logger(cls) -> HummingbotLogger:
     def __init__(self, trading_pairs: List[str]):
         super().__init__(trading_pairs)
 
+    @classmethod
+    async def init_trading_pair_symbols(cls, shared_session: Optional[aiohttp.ClientSession] = None):
+        """Initialize _trading_pair_symbol_map class variable
+        """
+        client_session = shared_session or aiohttp.ClientSession()
+        async with client_session as client:
+            async with client.get(BeaxyConstants.PublicApi.SYMBOLS_URL, timeout=5) as response:
+                if response.status == 200:
+                    symbols: List[Dict[str, Any]] = await response.json()
+                    cls._trading_pair_symbol_map = {
+                        symbol_data["name"]: (f"{symbol_data['baseCurrency']}-{symbol_data['termCurrency']}")
+                        for symbol_data in symbols
+                    }
+                else:
+                    IOError(f"There was an error requesting the list of symbols from the exchange ({response})")
+
+    @classmethod
+    async def trading_pair_symbol_map(cls) -> Dict[str, str]:
+        if not cls._trading_pair_symbol_map:
+            await cls.init_trading_pair_symbols()
+
+        return cls._trading_pair_symbol_map
+
+    @staticmethod
+    async def exchange_symbol_associated_to_pair(trading_pair: str) -> str:
+        symbol_map = await BeaxyAPIOrderBookDataSource.trading_pair_symbol_map()
+        symbols = [symbol for symbol, pair in symbol_map.items() if pair == trading_pair]
+
+        if symbols:
+            symbol = symbols[0]
+        else:
+            raise ValueError(f"There is no symbol mapping for trading pair {trading_pair}")
+
+        return symbol
```

Comment:
Given that this map will store all pairs from the exchange, I propose we use a two-way map to get this to O(1). [This answer](https://stackoverflow.com/a/13276237/6793798) has a nice implementation, except that I don't see why the `del`'s are needed, so I think this should be sufficient:
```Python
class TwoWayDict(dict):
    def __setitem__(self, key, value):
        dict.__setitem__(self, key, value)
        dict.__setitem__(self, value, key)

    def __delitem__(self, key):
        dict.__delitem__(self, self[key])
        dict.__delitem__(self, key)

    def __len__(self):
        """Returns the number of connections"""
        return dict.__len__(self) // 2
```